### PR TITLE
Sniper weapon tutorial

### PR DESCRIPTION
**Pull request by cdsupina**
I created this draft PR because I want to wait to write the section of the book until after we have settled on #116. The changes in this PR currently don't involve adding traits.

I instead generalized musket to "Gun" and added more struct values to customize how instances of a gun can work, number of bullets, shot speed, etc.

I then created a "spawn_musket" and "spawn_sniper" functions in their own item files.

Let me know what you think of this solution, I'm open to and expecting to make some changes based on your feedback.

### CODE REVIEW

**Comment by cdsupina** [comment]
Also, the sniper is in the test level, if you want to test it out. It has 2 bullets, faster shot speed, and more recoil than the musket.

**Review by olefasting — APPROVED** [review]

**Review by erlend-sh — APPROVED** [review]

File: assets/levels/lev02.json [hunk 1]
```diff
@@ -224,7 +224,7 @@
                 {
                  "height":0,
                  "id":56,
-                 "name":"muscet",
+                 "name":"musket",
```
Comment (grufkork) [inline assets/levels/lev02.json:227]: Best to rename it now before there's even more to do 👍

**Review by grufkork — COMMENTED** [review]
Looks good, very little duplicate code for the gun implementations. musket.rs and sniper.rs pretty much only contains data as of now, so that's a solid foundation for moving data to JSON or RON or whatever

**Comment by cdsupina** [comment]
I figured that perhaps in the future we can add traits like `Shootable` for shared behavior, but for now the sniper and musket basically have the exact same behavior, but with different values.

**Comment by cdsupina** [comment]
Adding a simple gun chapter is done! You can find it in book/src/simple_gun_weapon.md.

**Review by erlend-sh — APPROVED** [review]

File: book/src/simple_gun_weapon.md [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    fxses: &[],
+    network_ready: true,
+},
+```
+
+If you like, you can skip to the testing section to test your new weapon, but if you haven't added a new texture or modified an existing texture, you will not be able to visually recognize your item before picking it up. For this reason, I recommend you continue to the [Texture](#texture) section next.
+
+## Texture
+
+It is important to make sure that your weapon is able to be visually distinguished between the other weapons in the game. When I added the sniper rifle to the `ITEMS` array, I copied all of the data from existing musket weapon, changing only `tiled_name` and `constructor`. To give my sniper weapon a new texture, I will also need to change the "gun" texture in the `textures` field. Currently, the "gun" texture is set to `assets/Whale/Gun(92x32).png`. I'll open this file with my pixel editor of choice, [Aseprite](https://www.aseprite.org/) ([GIMP](https://www.gimp.org/) would also work fine).
```
Comment (grufkork) [inline book/src/simple_gun_weapon.md:146]: Could use an explanation for what `tiled_name` does (must match name in the map)

**Review by grufkork — APPROVED** [review]
Looks good! I suppose it could explain things more from the ground up but on the other hand, people who venture into gamedev with Rust probably are not entry-level 👍

File: book/src/simple_gun_weapon.md (hunk 2 above)
Comment (olefasting) [reply book/src/simple_gun_weapon.md:146]: 'tiled_name' should probably be changed to 'item_id' as we are moving away from Tiled. I haven't added items to the map editor yet, but when I do, I will probably have to change this, as well as create a way to define these in a data file, for reference by the editor and the scene builder

**Comment by cdsupina** [comment]
> Looks good! I suppose it could explain things more from the ground up but on the other hand, people who venture into gamedev with Rust probably are not entry-level 👍

I plan on adding or linking to resources for people more entry-level at some point. For example, how to clone, make a pull request, etc.

**Comment by cdsupina** [comment]
I clarified what the tiled_name field is. Merging now.

**Pull request merged**